packages/vite/src/node/plugins/resolve.ts

```diff
@@ -807,3 +859,10 @@ function mapWithBrowserField(
 function equalWithoutSuffix(path: string, key: string, suffix: string) {
   return key.endsWith(suffix) && key.slice(0, -suffix.length) === path
 }
+
+function getRealPath(resolved: string, preserveSymlinks?: boolean): string {
+  if (!preserveSymlinks && browserExternalId !== resolved) {
+    return (resolved = normalizePath(fs.realpathSync(resolved)))
```

Comment:
Is there a reason for the assignment in this line?
```suggestion
    return normalizePath(fs.realpathSync(resolved))
```

Reply:
Not noticed when refactoring the code 😅